linker/shared/src/main/scala/org/scalajs/linker/backend/emitter/NameCompressor.scala

```diff
@@ -0,0 +1,349 @@
+          if (!flags.isConstructor && !flags.isPrivate && !interfaceClassNames.contains(className))
+            registerPropertyName(methodName)
+
+        case UnaryOp(op, _) if !useBigIntForLongs =>
+          import UnaryOp._
+          (op: @switch) match {
+            case IntToLong    => registerPropertyName(LongImpl.fromInt)
+            case LongToInt    => registerPropertyName(LongImpl.toInt)
+            case LongToDouble => registerPropertyName(LongImpl.toDouble)
+            case DoubleToLong => registerPropertyName(LongImpl.fromDouble)
+            case LongToFloat  => registerPropertyName(LongImpl.toFloat)
+            case _            => ()
+          }
+
+        case BinaryOp(op, lhs, _) if !useBigIntForLongs =>
+          import BinaryOp._
+          (op: @switch) match {
+            case Long_+ => registerPropertyName(LongImpl.+)
+            case Long_- =>
+              lhs match {
+                case LongLiteral(0L) => registerPropertyName(LongImpl.UNARY_-)
+                case _               => registerPropertyName(LongImpl.-)
+              }
+            case Long_* => registerPropertyName(LongImpl.*)
+            case Long_/ => registerPropertyName(LongImpl./)
+            case Long_% => registerPropertyName(LongImpl.%)
+            case Long_| => registerPropertyName(LongImpl.|)
+            case Long_& => registerPropertyName(LongImpl.&)
+            case Long_^ =>
+              lhs match {
+                case LongLiteral(-1L) => registerPropertyName(LongImpl.UNARY_~)
+                case _                => registerPropertyName(LongImpl.^)
+              }
+            case Long_<<  => registerPropertyName(LongImpl.<<)
+            case Long_>>> => registerPropertyName(LongImpl.>>>)
+            case Long_>>  => registerPropertyName(LongImpl.>>)
+            case Long_==  => registerPropertyName(LongImpl.===)
+            case Long_!=  => registerPropertyName(LongImpl.!==)
+            case Long_<   => registerPropertyName(LongImpl.<)
+            case Long_<=  => registerPropertyName(LongImpl.<=)
+            case Long_>   => registerPropertyName(LongImpl.>)
+            case Long_>=  => registerPropertyName(LongImpl.>=)
+            case _        => ()
+          }
+
+        case Clone(expr) =>
+          if (expr.tpe.isInstanceOf[ArrayType])
+            registerPropertyName(cloneMethodName)
+
+        case UnwrapFromThrowable(expr) =>
+          registerPropertyName(JavaScriptExceptionClass, exceptionFieldName)
+
+        case JSGlobalRef(name) =>
+          registerGlobalRef(name)
+
+        case Transient(ZeroOf(_)) =>
+          registerPropertyName(ClassClass, dataFieldName)
+
+        case Transient(NativeArrayWrapper(elemClass, _)) =>
+          if (!elemClass.isInstanceOf[ClassOf])
+            registerPropertyName(ClassClass, dataFieldName)
+
+        case _ =>
+          ()
+      }
+    }
+  }
+
+  private final class NameGenerator(namesToAvoid: collection.Set[String]) {
```

Comment:
Write a unit test for this? IMO this would help a lot in understanding (and testing) of the sequences this generates. (and the test should be fairly simple to write).

Reply:
Done in the new `NameCompressorTest.scala`.